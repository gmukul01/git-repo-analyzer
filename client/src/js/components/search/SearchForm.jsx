import React, { PureComponent, Fragment } from "react";

export default class SearchForm extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      ownerName: "",
      repoName: ""
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }

  handleChange(evt) {
    this.setState({ [evt.target.name]: evt.target.value });
  }

  handleClick(evt) {
    evt.preventDefault();
    this.props.fetchRepoDetails(this.state);
  }

  render() {
    return (
      <Fragment>
        <form className="search--form">
          <input value={this.state.ownerName} onChange={this.handleChange} type="text" name="ownerName" className="form-control" placeholder="Owner Name" />
          <span>/</span>
          <input value={this.state.repoName} onChange={this.handleChange} type="text" name="repoName" className="form-control" placeholder="Repository Name" />
          <button className="btn btn-primary" onClick={this.handleClick}>
            Search
          </button>
        </form>
      </Fragment>
    );
  }
}
