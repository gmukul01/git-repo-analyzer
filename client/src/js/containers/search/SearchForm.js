import { connect } from "react-redux";

import SearchForm from "components/search/SearchForm";
import { fetchRepoDetails } from "actions/search";

const mapStateToProps = state => ({
  isLoading: state.search.isLoading,
  isFound: !state.search.errorMessage
});

export default connect(
  mapStateToProps,
  { fetchRepoDetails }
)(SearchForm);
