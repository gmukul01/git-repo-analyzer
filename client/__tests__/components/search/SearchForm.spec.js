import React from "react";
import { shallow } from "enzyme";
import ShallowRenderer from "react-test-renderer/shallow";

import SearchForm from "components/search/SearchForm";

describe("SearchForm page ", () => {
  let wrapper;
  const mockFetchRepoDetails = jest.fn();
  it("should match snapshot", () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<SearchForm fetchRepoDetails={mockFetchRepoDetails} />);
    expect(result).toMatchSnapshot();
  });

  beforeEach(() => {
    wrapper = shallow(<SearchForm fetchRepoDetails={mockFetchRepoDetails} />);
  });

  it("should call mockFetchRepoDetails on click of search button", () => {
    const evt = {
      preventDefault: jest.fn()
    };
    wrapper.find("button").simulate("click", evt);
    expect(mockFetchRepoDetails.mock.calls.length).toEqual(1);
    expect(evt.preventDefault.mock.calls.length).toEqual(1);
  });
});
