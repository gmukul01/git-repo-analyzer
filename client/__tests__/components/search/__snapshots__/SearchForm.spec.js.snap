// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm page  should match snapshot 1`] = `
<React.Fragment>
  <form
    className="search-form"
  >
    <input
      className="form-control"
      name="ownerName"
      onChange={[Function]}
      placeholder="Owner Name"
      type="text"
      value=""
    />
    <span>
      /
    </span>
    <input
      className="form-control"
      name="repoName"
      onChange={[Function]}
      placeholder="Repository Name"
      type="text"
      value=""
    />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Search
    </button>
  </form>
</React.Fragment>
`;
